### DISCUSSION TITLE
[Flutter] email verification & password reset

### DISCUSSION THREAD

**Discussion by goldcoders** [supabase/supabase#685] [Questions] [answered]
Ive tried flutter as the client for authentication with supabase.

Its cool i can login , sign up , signout.

i saw the packages in dart pub and tried the example.

my only problem is

i wanna make the behavious native, without going to web to verify email

same goes for password reset ...

I can make a web client deployed with netlify or vercel  to fetch act as the url 
when confirming email and resetting password.

but i think its possible to do also inside flutter mobile app?

is there a realtime subscription i can listen to?

lets say user click confirm password, i redirect user to another page instead of showing modal in the example.
then wait for the event when user verify the email.

then i can logged in the user.

same goes with reset,from password reset page, if they submit they will go to a page
where it is listening for an email verification.
then if user click the email they will be shown the change password page.

my only question is if i can use , dynamic url or firebase to handle the URL redirect to my page...
i havent tried it, so i need more insights if it is possible in this scenario that i dont wanna create a web client.

Or if you have an endpoint to acts as confirmation and password reset that would be awesome! 
:)

**Comment by kiwicopple** [comment]
Hey @goldcoders ! We're working on this :)

- https://github.com/supabase/gotrue/issues/62
- https://github.com/supabase/gotrue/issues/63

Once these are both in, we should be able to handle redirects to mobile applications within the libraries 👍

**Reply by atnegrete** [reply]
Looks like those are in, this is working I take it? Looks like both are merged & Site URL allows deep links.

**Comment by thorwebdev** [comment, marked as answer]
This is now documented here: https://supabase.io/docs/reference/dart/reset-password-email

And an example app can be found here: https://github.com/supabase/supabase/tree/master/examples/flutter-user-management#4-setup-deeplink-redirect-urls and with password reset here: https://github.com/phamhieu/supabase-flutter-demo
